geode-modules/src/main/java/org/apache/geode/services/module/impl/JBossModuleService.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.geode.services.module.ModuleDescriptor;
+import org.apache.geode.services.module.ModuleService;
+
+/**
+ * Implementation of {@link ModuleService} using JBoss-Modules.
+ */
+@Experimental
+public class JBossModuleService implements ModuleService {
+
+  private final Map<String, Module> modules = new HashMap<>();
+
+  private final GeodeModuleLoader moduleLoader = new GeodeModuleLoader();
+
+  private final Logger logger;
+
+  public JBossModuleService() {
+    this(LogService.getLogger());
+  }
+
+  public JBossModuleService(Logger logger) {
+    this.logger = logger;
+  }
+
+  public Module getModule(String name) {
+    return modules.get(name);
+  }
+
+  @Override
+  public boolean loadModule(ModuleDescriptor moduleDescriptor) {
+    logger.debug(String.format("Beginning to load module %s", moduleDescriptor.getVersionedName()));
+
+    if (modules.containsKey(moduleDescriptor.getVersionedName())) {
+      logger
+          .warn(String.format("Module %s is already loaded.", moduleDescriptor.getVersionedName()));
+      return false;
+    }
+
+    ModuleSpec.Builder builder = ModuleSpec.build(moduleDescriptor.getVersionedName());
+    builder.setVersion(Version.parse(moduleDescriptor.getVersion()));
+    builder.addDependency(new LocalDependencySpecBuilder()
+        .setImportServices(true)
+        .setExport(true)
+        .build());
+
+    moduleDescriptor.getDependedOnModules().forEach(dependency -> {
+      logger.debug(String.format("Adding dependency on module %s", dependency));
+      builder.addDependency(new ModuleDependencySpecBuilder()
+          .setName(dependency)
+          .build());
+    });
+
+    try {
+      for (String source : moduleDescriptor.getSources()) {
+        logger.debug(String.format("Adding resource %s to module", source));
+        ResourceLoader resourceLoader =
+            ResourceLoaders.createJarResourceLoader(new JarFile(source));
+        builder.addResourceRoot(ResourceLoaderSpec.createResourceLoaderSpec(resourceLoader));
+      }
+    } catch (IOException e) {
+      logger.error(e);
+      return false;
+    }
+
+    ModuleSpec moduleSpec = builder.create();
+    moduleLoader.addModuleSpec(moduleSpec);
+
+    try {
+      modules.put(moduleDescriptor.getVersionedName(),
+          moduleLoader.loadModule(moduleSpec.getName()));
+    } catch (ModuleLoadException e) {
+      logger.error(e);
+      return false;
+    }
+
+    logger
+        .debug(String.format("Module %s successfully loaded", moduleDescriptor.getVersionedName()));
+
+    return true;
+  }
+
+  @Override
+  public boolean unloadModule(String moduleName) {
+    return false;
+  }
+
+  @Override
+  public <T> List<T> loadService(Class<T> service) {
+    return null;
+  }
+
+  @Override
+  public <T> List<T> loadService(String serviceName) {
+    return null;
+  }
+
+  @Override
+  public boolean unloadService(String serviceName) {
+    return false;
+  }
```

Comment:
possibly we need to remove these add them into their respective JIRA

Reply:
Removed.